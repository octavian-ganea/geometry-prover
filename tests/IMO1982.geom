Description : IMO 1982 problem 2

Hypothesis:

Let an arbitrary circle Inscr( I ,radix)

Let an semiarbitrary point T1 situated on circle Inscr
Let an semiarbitrary point T2 situated on circle Inscr
Let an semiarbitrary point T3 situated on circle Inscr

Let a 2-equations deterministic point A1 situated on tangent at circle Inscr in point T2  and tangent at circle Inscr in point T3
Let a 2-equations deterministic point A2 situated on tangent at circle Inscr in point T1  and tangent at circle Inscr in point T3
Let a 2-equations deterministic point A3 situated on tangent at circle Inscr in point T1  and tangent at circle Inscr in point T2

Let a 1-equation deterministic point M1 situated on midpoint of A2 A3
Let a 1-equation deterministic point M2 situated on midpoint of A1 A3
Let a 1-equation deterministic point M3 situated on midpoint of A2 A1


Let a 1-equation deterministic point B1 situated on projection of T1 on line A1 I
Let a 1-equation deterministic point B2 situated on projection of T2 on line A2 I
Let a 1-equation deterministic point B3 situated on projection of T3 on line A3 I

Let a 1-equation deterministic point S1 situated on symmetric point of T1 with respect to B1
Let a 1-equation deterministic point S2 situated on symmetric point of T2 with respect to B2
Let a 1-equation deterministic point S3 situated on symmetric point of T3 with respect to B3

Conclusion:

concurrent  line M1 S1  line M2 S2  line M3 S3
