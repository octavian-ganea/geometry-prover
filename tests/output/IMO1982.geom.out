====== Parsing .... 
Description : IMO 1982 problem 2 

Hypothesis:

Let an arbitrary circle Inscr(I,radix)

Let a semiarbitrary point T1 situated on circle Inscr
Let a semiarbitrary point T2 situated on circle Inscr
Let a semiarbitrary point T3 situated on circle Inscr

Let a 2-equations deterministic point A1 situated on tangent at circle Inscr in point T2 and tangent at circle Inscr in point T3
Let a 2-equations deterministic point A2 situated on tangent at circle Inscr in point T1 and tangent at circle Inscr in point T3
Let a 2-equations deterministic point A3 situated on tangent at circle Inscr in point T1 and tangent at circle Inscr in point T2
Let a 1-equation deterministic point M1 situated on midpoint of A2 A3
Let a 1-equation deterministic point M2 situated on midpoint of A1 A3
Let a 1-equation deterministic point M3 situated on midpoint of A2 A1
Let a 1-equation deterministic point B1 situated on projection of T1 on line A1-I
Let a 1-equation deterministic point B2 situated on projection of T2 on line A2-I
Let a 1-equation deterministic point B3 situated on projection of T3 on line A3-I
Let a 1-equation deterministic point S1 situated on symmetric point of T1 with respect to B1
Let a 1-equation deterministic point S2 situated on symmetric point of T2 with respect to B2
Let a 1-equation deterministic point S3 situated on symmetric point of T3 with respect to B3
Let a 2-equations deterministic point X13 situated on line M1-S1 and line M2-S2

Conclusion: points X13, M3 and S3 are collinear

====== Parse OK 

====== Proof of the problem:

======= Generating equations for each point ... 

========= List of arbitrary points:


========= List of semi-arbitrary points:

T1 is defined by equation: t1 * t1__ - 1 = 0
After substitutions and simplifications:
 t1__ = 1  / t1

T2 is defined by equation: t2 * t2__ - 1 = 0
After substitutions and simplifications:
 t2__ = 1  / t2

T3 is defined by equation: t3 * t3__ - 1 = 0
After substitutions and simplifications:
 t3__ = 1  / t3

========= List of deterministic points:

A1 is defined by equation: a1 + a1__ * t2^2 - 2 * t2 = 0
         and by equation: a1 + a1__ * t3^2 - 2 * t3 = 0
After substitutions and simplifications:
 a1  =  2 * t2 * t3  / t3 + t2
 a1__  =  2  / t2 + t3 


A2 is defined by equation: a2 + a2__ * t1^2 - 2 * t1 = 0
         and by equation: a2 + a2__ * t3^2 - 2 * t3 = 0
After substitutions and simplifications:
 a2  =  2 * t1 * t3  / t3 + t1
 a2__  =  2  / t1 + t3 


A3 is defined by equation: a3 + a3__ * t1^2 - 2 * t1 = 0
         and by equation: a3 + a3__ * t2^2 - 2 * t2 = 0
After substitutions and simplifications:
 a3  =  2 * t1 * t2  / t2 + t1
 a3__  =  2  / t1 + t2 


M1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 m1  =  2 * t1 * t2 * t3 + t1^2 * t3 + t1^2 * t2  / t2 * t3 + t1 * t2 + t1 * t3 + t1^2 

 m1__  =  2 * t1 + t2 + t3  / t1^2 + t1 * t3 + t1 * t2 + t2 * t3 


M2 is a deterministic point defined by equations (After substitutions and simplifications):
 
 m2  =  t2^2 * t3 + 2 * t1 * t2 * t3 + t1 * t2^2  / t2 * t3 + t2^2 + t1 * t3 + t1 * t2 

 m2__  =  t1 + 2 * t2 + t3  / t1 * t2 + t1 * t3 + t2^2 + t2 * t3 


M3 is a deterministic point defined by equations (After substitutions and simplifications):
 
 m3  =  t1 * t3^2 + 2 * t1 * t2 * t3 + t2 * t3^2  / t3^2 + t2 * t3 + t1 * t3 + t1 * t2 

 m3__  =  t2 + 2 * t3 + t1  / t1 * t2 + t1 * t3 + t2 * t3 + t3^2 


B1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 b1  =  t2 * t3 + t1^2  / 2 * t1 

 b1__  =  t1^2 + t2 * t3  / 2 * t1 * t2 * t3 


B2 is a deterministic point defined by equations (After substitutions and simplifications):
 
 b2  =  t1 * t3 + t2^2  / 2 * t2 

 b2__  =  t2^2 + t1 * t3  / 2 * t1 * t2 * t3 


B3 is a deterministic point defined by equations (After substitutions and simplifications):
 
 b3  =  t1 * t2 + t3^2  / 2 * t3 

 b3__  =  t3^2 + t1 * t2  / 2 * t1 * t2 * t3 


S1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 s1  =  t2 * t3  / t1 

 s1__  =  t1  / t2 * t3 


S2 is a deterministic point defined by equations (After substitutions and simplifications):
 
 s2  =  t1 * t3  / t2 

 s2__  =  t2  / t1 * t3 


S3 is a deterministic point defined by equations (After substitutions and simplifications):
 
 s3  =  t1 * t2  / t3 

 s3__  =  t3  / t1 * t2 


X13 is defined by equation: m1__ * x13 + m1 * s1__ + s1 * x13__ - s1__ * x13 - m1 * x13__ - m1__ * s1 = 0
         and by equation: m2__ * x13 + m2 * s2__ + s2 * x13__ - s2__ * x13 - m2 * x13__ - m2__ * s2 = 0
After substitutions and simplifications:
 x13  =  3 * t1^3 * t2^3 * t3^2 + t1^3 * t2^2 * t3^3 + t1^2 * t2^3 * t3^3 + t1 * t2^3 * t3^4 + 2 * t1^4 * t2^3 * t3 + t1^4 * t2^4 + 2 * t1^3 * t2^4 * t3 + 2 * t1^2 * t2^2 * t3^4 + t1^3 * t2 * t3^4 - 2 * t1^5 * t2 * t3^2 - t1^5 * t2^2 * t3 - t1^5 * t3^3 - t1^2 * t2^4 * t3^2 - 2 * t1 * t2^4 * t3^3 - t1^2 * t2^5 * t3 - 2 * t1 * t2^5 * t3^2 - t2^5 * t3^3 - 2 * t1^4 * t2 * t3^3 - t1^4 * t2^2 * t3^2  / t1^3 * t2^2 * t3^2 + 3 * t1^3 * t2^3 * t3 + t1^2 * t2^3 * t3^2 + 3 * t1^2 * t2^2 * t3^3 + t1^4 * t2^3 + t1^3 * t2^4 + t1^2 * t2 * t3^4 + t1 * t2^2 * t3^4 - t1^5 * t3^2 - t1^4 * t3^3 - t1^5 * t2 * t3 - 3 * t1 * t2^4 * t3^2 - t1 * t2^5 * t3 - t2^5 * t3^2 - t2^4 * t3^3 - 3 * t1^4 * t2 * t3^2
 x13__  =  3 * t1^2 * t2^2 * t3^2 + t1^2 * t2^3 * t3 + t1^3 * t2^2 * t3 + t1^4 * t2^2 + 2 * t1 * t2^2 * t3^3 + t1 * t2 * t3^4 + 2 * t1^2 * t2 * t3^3 + 2 * t1^3 * t2^3 + t1^2 * t2^4 - 2 * t2^4 * t3^2 - t2^3 * t3^3 - t2^5 * t3 - t1^3 * t2 * t3^2 - 2 * t1^4 * t2 * t3 - t1^3 * t3^3 - 2 * t1^4 * t3^2 - t1^5 * t3 - 2 * t1 * t2^4 * t3 - t1 * t2^3 * t3^2  / t1^2 * t2^3 * t3^2 + 3 * t1^2 * t2^2 * t3^3 + t1^3 * t2^2 * t3^2 + 3 * t1^3 * t2^3 * t3 + t1 * t2^2 * t3^4 + t1^2 * t2 * t3^4 + t1^3 * t2^4 + t1^4 * t2^3 - t2^5 * t3^2 - t1 * t2^5 * t3 - t2^4 * t3^3 - 3 * t1^4 * t2 * t3^2 - t1^4 * t3^3 - t1^5 * t3^2 - t1^5 * t2 * t3 - 3 * t1 * t2^4 * t3^2 

=========== Equations were generated successfully ========

======= The conclusion may be correct only unde the following restrictions:
 -- points T2 and T1 can not be symmetric with respect to the center of the arbitrary circle 
 -- points I and A3 must be different
 -- points T3 and T1 can not be symmetric with respect to the center of the arbitrary circle 
 -- points I and A2 must be different
 -- points T3 and T2 can not be symmetric with respect to the center of the arbitrary circle 
 -- points I and A1 must be different
 -- points T2 and T1 must be different
 -- points T3 and T1 must be different
 -- points T3 and T2 must be different

====== Trying to solve the goal of the problem ...
Conclusion is equivalent to equation: 
m3__ * x13 + m3 * s3__ + s3 * x13__ - s3__ * x13 - m3 * x13__ - m3__ * s3 = 0

Attempt to prove it by substitutions and simplifications ....
 
  !!!!!!!!!!!!!!!!!!!!!
  !!!!!! SUCCESS !!!!!!
  !!!!!!!!!!!!!!!!!!!!!

===== Problem is correct. Proof ends here. QED ====

